src/show.jl

```diff
@@ -0,0 +1,224 @@
+function Base.show(io::IO, m::Model)
+    println(io, "MPSGE model with $(length(m._sectors)) sectors, $(length(m._commodities)) commodities and $(length(m._consumers)) consumers.")
+
+    if length(m._sectors) > 0
+        print(io, "  Sectors: ")
+        print(io, join(["$(s.name) (bm=$(s.benchmark))" for s in m._sectors], ", "))
+        println(io)
+    end
+
+    if length(m._commodities) > 0
+        print(io, "  Commodities: ")
+        print(io, join(["$(c.name) (bm=$(c.benchmark))" for c in m._commodities], ", "))
+        println(io)
+    end
+
+    if length(m._consumers) > 0
+        print(io, "  Consumers: ")
+        print(io, join(["$(c.name) (bm=$(c.benchmark))" for c in m._consumers], ", "))
+        println(io)
+    end
+
+    if length(m._auxs) > 0
+        print(io, "  Auxs: ")
+        print(io, join(["$(a.name) (bm=$(a.benchmark))" for a in m._auxs], ", "))
+        println(io)
+    end
+
+    if m._jump_model!==nothing
+        if m._status==:Solved
+            println(io, "Solution:")
+
+            for n in JuMP.all_variables(m._jump_model)
+                println(io, "  $n:\t$(Complementarity.result_value(n))")
+            end        
+        else
+            println(io, "Did not solve with error: $(m._status).")
+        end
+    end
+end
+
+
+
+function Base.show(io::IO,S::Union{SectorRef,CommodityRef,ConsumerRef,AuxRef,ParameterRef})
+    #print(io,get_full(S))
+    if isnothing(S.subindex)
+        return print(io,get_full(S))
+    else
+        names = S.subindex_names
+        full_S = get_full(S)
+        return print(io,"$(names)\tbm: $(full_S.benchmark[names...])")
+    end
+
+end
```

Comment:
I've solve the issue of the dense axis arrays not printing correctly is a fairly hacky way. I check if the SectorRef refers to an IndexSector by testing if the subindex is nothing. It's not a great solution.